tests/parse_shebang_test.py

```diff
@@ -88,7 +88,21 @@ def test_normexe_does_not_exist():
 def test_normexe_does_not_exist_sep():
     with pytest.raises(OSError) as excinfo:
         parse_shebang.normexe('./i-dont-exist-lol')
-    assert excinfo.value.args == ('Executable `./i-dont-exist-lol` not found',)
+    assert excinfo.value.args == (
+        'Executable `./i-dont-exist-lol` not found',
+    )
+
+
+def test_normexe_not_executable(in_tmpdir):
+    exe_path = os.path.abspath(
+        write_executable('/usr/bin/env sh', filename='run.myext'),
+    )
+    os.chmod(exe_path, 444)
+    with pytest.raises(OSError) as excinfo:
+        parse_shebang.normexe(exe_path)
+    assert excinfo.value.args == (
+        'Executable `{}` is not executable'.format(exe_path),
+    )
```

Comment:
I suspect you might need to skip this test on windows given the way executables work there

you can use `# pragma: windows no cover` this test (and probably the code branch as well if it isn't going to be hit)